modules/phase_field/src/kernels/ACGrGrPolyStoredEnergy.C

```diff
@@ -0,0 +1,58 @@
+/****************************************************************/
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*          All contents are licensed under LGPL V2.1           */
+/*             See LICENSE for full restrictions                */
+/****************************************************************/
+#include "ACGrGrPolyStoredEnergy.h"
+
+template<>
+InputParameters validParams<ACGrGrPolyStoredEnergy>()
+{
+  InputParameters params = validParams<ACBulk>();
+  params.addClassDescription("Stored energy contribution for the poly crystaline Allen-Cahn Kernel");
+  params.addRequiredParam<std::vector<Real> >("stored_energy", "list of stored energies for each grain in the simulation");
```

Comment:
Is this the best interface for this? If you have a 1000 grain simulation, which isn't unusual, you would have to have a vector with 1000 entries